rdrf/rdrf/security_checks.py

```diff
@@ -0,0 +1,66 @@
+from registry.patients.models import Patient
+from registry.patients.models import ParentGuardian
+from django.core.exceptions import PermissionDenied
+import logging
+
+logger = logging.getLogger(__name__)
+
+def _get_prop(user, prop):
+    # case where site does not have prop defined
+    try:
+        return getattr(user, prop)
+    except:
+        return False
+
+def _user_is_patient_type(user):
+    return any([_get_prop(user, "is_patient"),
+                _get_prop(user, "is_parent"),
+                _get_prop(user, "is_carrier")])
+
+
+def _security_violation(user, patient_model):
```

Comment:
Call this _raise_security_violation to make the code below more obvious where the function exits.